Shopify/cli
# Remove theme directory confirmation during tests and make confirmation dialogue respect `SHOPIFY_CLI_TTY`

### WHY are these changes introduced?

Fixes #1136 

#### Issue 1: `@ctx.testing?` is `true` even outside a testing environment

CLI was hanging when `theme pull` was executed in a directory with already existing files.
CLI should have shown question

> It doesn’t seem like you’re running this command in a theme directory. Are you sure you want to proceed?

but was never shown, due to

```ruby
raise "Current theme directory can't be confirmed during tests" if @ctx.testing?
```

Discussion why `raise … if @ctx.testing?` was introduced can be found [here](https://github.com/Shopify/shopify-cli/pull/2680#discussion_r1018371466). Because `@ctx.testing?` returns `true` based on `ENV["CI"]`, we can’t differentiate between CLI running in partner’s pipeline or in CLI tests.

#### Issue 2: Confirmation dialogue ignores environment variable `SHOPIFY_CLI_TTY`

Moreover, even after having removed `raise … if @ctx.testing?` and `SHOPIFY_CLI_TTY` being set to `0`, CLI would still hang at the “It doesn’t seem like …” question.

### WHAT is this pull request doing?

1. Removes `raise … if @ctx.testing?`.
2. Shows a warning `It doesn’t seem like you’re running this command in a theme directory.` when `SHOPIFY_CLI_TTY` is set to `0`, but continues to `pull`.
    * `--force` flag as a way to confirm the command will not be included in the error message yet.
      * → https://github.com/Shopify/shopify-cli/pull/2680#discussion_r1017721847
      * → https://github.com/Shopify/cli/issues/760

### How to test your changes?

1. `mkdir ./test-folder`
2. `cd ./test-folder`
4. `touch randomfile1.txt randomfile2.log`
5. `cd ..`
6. Retrieve a random theme ID from `shopify theme list`
7. Run `CI=true pnpm shopify:run theme pull --theme <Your Theme ID> --path="./test-folder"`.
    * Verify that warning `It doesn’t seem like you’re running this command in a theme directory.` and confirmation dialogue `Are you sure you want to proceed?` are shown.
    * `Ctrl+C`.
9. Run `SHOPIFY_CLI_TTY=0 CI=true pnpm shopify:run theme pull --theme <Your Theme ID> --path="./test-folder"`.
    * Verify that CLI displays warning `It doesn’t seem like you’re running this command in a theme directory.` but continues to `pull`.
    * `Ctrl+C` the pull and make sure that your test-folder is a non-theme directory (again).
11. Run `CI=true pnpm shopify:run theme pull --theme <Your Theme ID> --path="./test-folder" --force`.
    * Verify that neither warning nor confirmation dialogue are shown and that theme files get pulled.

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
